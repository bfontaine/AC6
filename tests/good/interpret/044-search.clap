def search (start) (stop) (accu) (what) = 
  if ~ (start = stop + 1) then 
    ((what start accu) . case {
      Continue [accu] => search (start + 1) stop accu what
    | Success [accu] => Success [accu]
    | Fail => Fail
    })
  else 
    Fail

def pred (l) (h) (s) = 
  fun (k) (accu) => 
    if accu = s then 
      Success [accu] 
    else if (l <= k) && (k <= h) then 
      Continue [accu + k] 
    else Fail

val r1 = search 1 10 0 (pred 1 5 6)

val r2 = search 1 10 0 (pred 3 7 42)

