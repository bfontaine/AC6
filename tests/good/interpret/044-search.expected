:- search = <code>
:- pred = <code>
:- r1 = Success [6]
:- r2 = Fail
